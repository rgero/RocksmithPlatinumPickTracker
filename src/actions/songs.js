
export const addSong = (song) => ({
    type: 'ADD_SONG',
    song
})

export const removeSong = ( id ) => ({
    type: 'REMOVE_SONG',
    id
})

export const editSong = ( id, update = {}) => ({
    type: "EDIT_SONG",
    id,
    update
})

// SET_SONG
export const setSongs = (songs) => ({
    type: 'SET_SONGS',
    songs
});