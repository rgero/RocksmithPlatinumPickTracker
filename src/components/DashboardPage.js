import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import PickListFilter from './PickListFilters';
import PickList from './PickList';




const DashboardPage = () => (
    (
        <div>
            <div className='content-container'>
                <div className='content-navbar'>
                    <Link className='button' to='/import'>Import Data</Link>
                    <Link className='button' to='/create'>Log New Pick</Link>
                </div>
                <PickListFilter />
                <PickList />
            </div>
        </div>
    )
);

export default DashboardPage;