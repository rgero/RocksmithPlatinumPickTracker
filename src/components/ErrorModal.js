import React from 'react';
import Modal from 'react-modal';

const ErrorModal = (props) =>
    (
        <Modal
            isOpen={!!props.errorMessage}
            contentLabel='Error with Form' // This is for people who have accessiblity options enabled.
            onRequestClose={props.clearError} // This is fired if you click in the background or the escape key.
            closeTimeoutMS={200}
            className='modal'
        >
            {props.errorMessage && 
                <div>
                    <h2>Error</h2>
                    <div>{props.errorMessage}</div>
                    <div><button className='button' onClick={props.clearError}>Close</button></div>
                </div>
            }
        </Modal>
    )

export default ErrorModal