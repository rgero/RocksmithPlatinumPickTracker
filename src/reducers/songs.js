// This is the songs reducer
const songsReducerDefaultState = [];
const songsReducer = (state = songsReducerDefaultState, action) => {
    switch(action.type){
        case "ADD_SONG":
            // Get the highest ID - this is temporary until I set up firebase.
            var songsLength = state.length;
            action.song.id = (songsLength+1).toString();
            
            return state.concat(action.song);
        case "REMOVE_SONG":
            return state.filter(function(testOption){
                return action.id !== testOption.id;
            })
        case "EDIT_SONG":
            return state.map( (song) => {
                if (song.id === action.id) {
                    return {
                        ...song,
                        ...action.update
                    }
                } else {
                    return song;
                }
            })
        case 'SET_SONGS':
            return action.songs;
        default:
            return state;
    }
};

export default songsReducer;