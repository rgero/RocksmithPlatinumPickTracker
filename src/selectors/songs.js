// How we parse the data.
import moment from 'moment';

// Get visible songs
const getVisibleSongs = (songs, { text, startDate, endDate }) => {
    return songs.filter((song)=> {
        const createdMoment = moment(song.date);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdMoment, 'day') : true;
        const textMatch = typeof text !== "string" || song.artist.toLowerCase().includes(text.toLowerCase()) || song.songName.toLowerCase().includes(text.toLowerCase())

        return startDateMatch && endDateMatch && textMatch;
    })
}

export {getVisibleSongs}