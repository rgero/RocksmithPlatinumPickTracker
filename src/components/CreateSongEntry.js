import React from 'react';
import {connect} from 'react-redux';
import SongForm from './SongForm';
import { addSong } from '../actions/songs';

export class CreateSongEntry extends React.Component {
    constructor(props){
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
    }

    onSubmit(song){
        this.props.onSubmit(song);
        this.props.history.push('/')
    }

    render(){
        return(
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Song</h1>
                    </div>
                </div>
                <div className="content-container">
                    <SongForm onSubmit={this.onSubmit} />
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (song) => dispatch(addSong(song))
    }
}

export default connect(undefined, mapDispatchToProps)(CreateSongEntry); // Check out the react-redux documentation to understand the connect statement here.