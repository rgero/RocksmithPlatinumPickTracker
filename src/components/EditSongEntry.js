import React from 'react';
import {connect} from 'react-redux';
import SongForm from './SongForm';
import {startEditSong, startRemoveSong} from '../actions/songs';

export class EditSongPage extends React.Component {
    constructor(props){
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
        this.removeItem = this.removeItem.bind(this);
    }

    onSubmit(song){
        this.props.onSubmit(this.props.song.id, song)
        this.props.history.push('/')
    }

    removeItem(){
        this.props.removeItem(this.props.song.id);
        this.props.history.push('/')
    }

    render(){
        return(
            <div>
                <div className='page-header'>
                    <div className='content-container'>
                        <h1 className='page-header__title'>Editing '{this.props.song.artist} - {this.props.song.songName}'</h1>
                    </div>
                </div>
                <div className='content-container'>
                    <SongForm
                        songInfo = {this.props.song}
                        onSubmit={this.onSubmit}
                        
                    />
                    <button className='button button--secondary' onClick={this.removeItem}>Remove</button>
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (id, song) => { 
            dispatch(startEditSong(id, song))
        },
        removeItem: (id) => dispatch(startRemoveSong(id))
    }
}

const mapStateToProps = (state, props) => {
    return {
        song: state.songs.find((song) => song.id === props.match.params.id)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditSongPage);