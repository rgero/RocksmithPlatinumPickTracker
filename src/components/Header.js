import React from 'react';
import { Link } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="header__content">
            <Link className="header__title" to="/practices" exact="true"><h1>Rocksmith Pick Tracker</h1></Link>
            <button className="button button--link" onClick={startLogout}>Log out</button>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({

    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);