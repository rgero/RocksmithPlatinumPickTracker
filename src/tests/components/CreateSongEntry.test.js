import React from 'react';
import {shallow} from 'enzyme';
import {CreateSongEntry} from '../../components/CreateSongEntry';

import songs from '../fixtures/songs';

let onSubmit, history, wrapper;

beforeEach(()=>{
    onSubmit = jest.fn();
    history = { push: jest.fn() }
    wrapper = shallow(<CreateSongEntry onSubmit={onSubmit} history={history}/>)
})

test("Snapshot test of the rendering", ()=>{
    expect(wrapper).toMatchSnapshot();
})

test('OnSubmit testing', ()=> {
    const testObj = wrapper.find('SongForm');
    expect(testObj.length).toBe(1);
    testObj.prop('onSubmit')(songs[0]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(onSubmit).toHaveBeenLastCalledWith(songs[0]);
})