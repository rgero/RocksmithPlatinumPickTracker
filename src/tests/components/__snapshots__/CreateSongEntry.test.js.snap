// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of the rendering 1`] = `
<div>
  <Header />
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Add Song
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <SongForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
